src/storage/mod.rs

```diff
@@ -919,6 +919,60 @@ impl Storage {
             .flatten()
     }
 
+    pub fn async_raw_batch_get(
+        &self,
+        ctx: Context,
+        keys: Vec<Vec<u8>>,
+    ) -> impl Future<Item = Vec<Result<KvPair>>, Error = Error> {
+        const CMD: &str = "raw_batch_get";
+        let engine = self.get_engine();
+        let priority = readpool::Priority::from(ctx.get_priority());
+
+        let keys: Vec<Key> = keys.into_iter().map(Key::from_encoded).collect();
+
+        let res = self.read_pool.future_execute(priority, move |ctxd| {
+            let mut _timer = {
+                let ctxd = ctxd.clone();
+                let mut thread_ctx = ctxd.current_thread_context_mut();
+                thread_ctx.start_command_duration_timer(CMD, priority, true)
+            };
+
+            Self::async_snapshot(engine, &ctx)
+                .and_then(move |snapshot: Box<Snapshot>| {
+                    let mut thread_ctx = ctxd.current_thread_context_mut();
+                    let _t_process = thread_ctx.start_processing_read_duration_timer(CMD);
+
+                    // no scan_count for this kind of op.
+                    let mut stats = Statistics::default();
+                    thread_ctx.collect_key_reads(CMD, keys.len() as u64);
+                    let result: Vec<Result<KvPair>> = keys.iter()
+                        .map(|k| (k, snapshot.get(k)))
+                        .filter(|&(_, ref v)| !(v.is_ok() && v.as_ref().unwrap().is_none()))
+                        .into_iter()
+                        .map(|(k, v)| match v {
+                            Ok(Some(v)) => {
+                                stats.data.flow_stats.read_keys += 1;
+                                stats.data.flow_stats.read_bytes = k.encoded().len() + v.len();
+                                Ok((k.encoded().clone(), v))
```

Comment:
Use `.raw().unwrap()` like `async_batch_get`?

Reply:
Oops. Since you mentioned async_batch_get. If I understand it correctly, raw key requires encoded family method. Therefore the .raw().unwrap() in async_batch_get is potentially incorrect, am I right?

Reply:
Just checked, the async_batch_get is for kv interface instead of rawkv interface. And it is not related to this merge request. And I still assume that in rawkv interface, key should use encoded family of methods instead.